gcore/src/exec.rs

```diff
@@ -115,7 +115,7 @@ pub fn block_timestamp() -> u64 {
 /// ```
 pub fn reply_deposit(message_id: MessageId, amount: u64) -> Result<()> {
     let mut error_code = 0u32;
-    unsafe { gsys::gr_reply_deposit(message_id.as_ptr(), amount, &mut error_code) };
+    unsafe { gsys::gr_reply_deposit(message_id.as_ref().as_ptr() as _, amount, &mut error_code) };
```

Comment:
Consider to introduce `as_ptr` for `gprimitives` types

Reply:
I see no reason to provide raw ptrs API foruserspace library

Reply:
We can introduce `MessageIdExt` for `gcore` with `as_ptr` and `as_mut_ptr`.